src/core/internal/traits.d

```diff
@@ -277,14 +277,51 @@ template hasElaborateCopyConstructor(S)
     }
     else static if (is(S == struct))
     {
-        enum hasElaborateCopyConstructor = __traits(hasMember, S, "__xpostblit");
+        enum hasElaborateCopyConstructor = __traits(hasMember, S, "__xpostblit") ||
+             () {
+                static if (__traits(hasMember, S, "__ctor"))
+                {
+                    static foreach (f; __traits(getOverloads, S, "__ctor"))
+                    {{
+                        static if (Parameters!f.length == 1)
```

Comment:
Copy constructors may also have any number of default parameters.

Reply:
How about just
```
__traits(compiles, { auto r = U.init; U copy = r; })
```

instead of all this?

Reply:
Nevermind, could be non-trivial.

Reply:
This won’t work as it doesn’t rule out blitting.